core/models.py

```diff
@@ -213,19 +213,24 @@ def apply_ordering(self, query):
         return qs
 
     def filter_by_querystring(self, querystring):
-        queryset = self
+        query, search_query, order_by = self.parse_querystring(querystring)
+        return self.composed_query(query, search_query, order_by)
+
+    def parse_querystring(self, querystring):
         query = querystring.copy()
         order_by = query.pop("order-by", [""])
         order_by = [field.strip().lower() for field in order_by[0].split(",") if field.strip()]
         search_query = query.pop("search", [""])[0]
         query = {key: value for key, value in query.items() if value}
-        if search_query:
-            queryset = queryset.search(search_query)
-        if query:
-            queryset = queryset.apply_filters(query)
-        queryset = queryset.apply_ordering(order_by)
+        return query, search_query, order_by
 
-        return queryset
+    def composed_query(self, filter_query=None, search_query=None, order_by=None):
+        qs = self
+        if filter_query:
+            qs = qs.apply_filters(filter_query)
+        if search_query:
+            qs = qs.search(search_query)
+        return qs.apply_ordering(order_by or [""])
```

Comment:
Acredito que aqui deveria ser:
```suggestion
        return qs.apply_ordering(order_by or [])
```
Não?